src/utils.jl

```diff
@@ -56,11 +58,51 @@ end
 
 # family of projection functions
 
-immutable Proj{field}; end
+immutable Proj{F}
+    f::F
```

Comment:
I find this definition very confusing. A `Proj{f}()` is supposed to be a function that selects field `f`, period. Anything else should be a different type.

Also I realize the definition `(p::Proj)(c::Columns)` isn't really right anymore, and should be a method of broadcast now.